programs/rodeos/ship_client.hpp

```diff
@@ -185,6 +187,14 @@ struct connection : connection_base {
    void catch_and_close(F f) {
       try {
          f();
+      } catch (const std::runtime_error& e) {
+         elog("${e}", ("e", e.what()));
+         
+         if (std::strstr(e.what(), "quit rodeos") != nullptr) {
```

Comment:
Do this a diff way. Can use EOS exceptions like everywhere else in our code.

Reply:
Thanks for pointing this out. I did not like strstr either and was trying to follow the original throw style. Changed to `eosio::chain::unlinkable_block_exception`.